pkg/cli/commands.go

```diff
@@ -516,9 +520,17 @@ func MakeGRPCServerCommand[T field.Configurable](
 			}
 			runCtx = context.WithValue(runCtx, crypto.ContextClientSecretKey, secretJwk)
 		}
+
+		sessionStoreMaximumSize := v.GetInt(field.ServerSessionStoreMaximumSizeField.GetName())
 		sessionConstructor := getGRPCSessionStoreClient(runCtx, serverCfg)
 		c, err := getconnector(runCtx, t, RunTimeOpts{
-			SessionStore: &lazySessionStore{constructor: sessionConstructor},
+			SessionStore: NewLazyCachingSessionStore(sessionConstructor, func(otterOptions *otter.Options[string, []byte]) {
```

Comment:
This needs further thinking. The cache can be preserved between syncs.  Maybe we just namespace everything in one otter via a sync_id prefix?